File: Models/Tenant/Project/Job/JobStatus.cs
using System;
using System.Collections.Generic;
using Warehouse.Models.Joins;

namespace Warehouse.Models
{
    public class JobStatus
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Colour { get; set; }
        public IList<Job> Jobs { get; set; }
    }

    public class NewStatus
    {
        public string Name { get; set; }
        public string Colour { get; set; }
    }

    public class JobStatusViewModel
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Colour { get; set; }
    }
}

File: Warehouse.Web/ClientApp/src/app/Models/JobStatus.ts
import {User} from "./User";

export interface JobStatus {
    id: string;
    name: string;
    colour: string;
    finished: boolean;
    order: number;
    user: User | null;
}

export interface NewStatus
{
  name: string;
  order: number;
  colour: string;
  finished: boolean;
}
